### PR TITLE
JDK-8301074: Replace NULL with nullptr in share/opto/

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory share/opto/. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

    No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
    Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
    nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how nullptr participates in a code expression here, we really are talking about the specific value nullptr.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301074](https://bugs.openjdk.org/browse/JDK-8301074): Replace NULL with nullptr in share/opto/

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [41bb3390](https://git.openjdk.org/jdk/pull/12187/files/41bb33905a2c052509b9ad207e3fb36f7e6c7126)
 * [Jesper Wilhelmsson](https://openjdk.org/census#jwilhelm) (@JesperIRL - **Reviewer**) ⚠️ Review applies to [10902048](https://git.openjdk.org/jdk/pull/12187/files/1090204833cebde62c3b70e74c832a037f126de1)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12187/head:pull/12187` \
`$ git checkout pull/12187`

Update a local copy of the PR: \
`$ git checkout pull/12187` \
`$ git pull https://git.openjdk.org/jdk pull/12187/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12187`

View PR using the GUI difftool: \
`$ git pr show -t 12187`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12187.diff">https://git.openjdk.org/jdk/pull/12187.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/node.cpp [hunk 1]
```diff
@@ -776,19 +776,19 @@ void Node::add_req( Node *n ) {
...
+  if( (_cnt >= _max) || (in(_max-1) != nullptr) )
     grow( _max+1 );
 
   // Find a precedence edge to move
-  if( in(_cnt) != NULL ) {       // Next precedence edge is busy?
+  if( in(_cnt) != nullptr ) {       // Next precedence edge is busy?
     uint i;
     for( i=_cnt; i<_max; i++ )
-      if( in(i) == NULL )       // Find the NULL at end of prec edge list
+      if( in(i) == nullptr )       // Find the null at end of prec edge list
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.cpp:786]: align

File: src/hotspot/share/opto/node.cpp [hunk 2]
```diff
@@ -776,19 +776,19 @@ void Node::add_req( Node *n ) {
   assert( is_not_dead(n), "can not use dead node");
 
   // Look to see if I can move precedence down one without reallocating
-  if( (_cnt >= _max) || (in(_max-1) != NULL) )
+  if( (_cnt >= _max) || (in(_max-1) != nullptr) )
     grow( _max+1 );
 
   // Find a precedence edge to move
-  if( in(_cnt) != NULL ) {       // Next precedence edge is busy?
+  if( in(_cnt) != nullptr ) {       // Next precedence edge is busy?
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.cpp:783]: align

File: src/hotspot/share/opto/node.cpp [hunk 3]
```diff
@@ -808,10 +808,10 @@ void Node::add_req_batch( Node *n, uint m ) {
     grow( _max+m );
 
   // Find a precedence edge to move
-  if( _in[_cnt] != NULL ) {     // Next precedence edge is busy?
+  if( _in[_cnt] != nullptr ) {     // Next precedence edge is busy?
     uint i;
     for( i=_cnt; i<_max; i++ )
-      if( _in[i] == NULL )      // Find the NULL at end of prec edge list
+      if( _in[i] == nullptr )      // Find the null at end of prec edge list
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.cpp:814]: align

File: src/hotspot/share/opto/node.cpp [hunk 4]
```diff
@@ -2070,10 +2070,10 @@ void PrintBFS::print_options_help(bool print_examples) {
   tty->print("Arguments:\n");
   tty->print("  this/start: staring point of BFS\n");
   tty->print("  target:\n");
-  tty->print("    if nullptr: simple BFS\n");
+  tty->print("    if null: simple BFS\n");
   tty->print("    else: shortest path or all paths between this/start and target\n");
   tty->print("  options:\n");
-  tty->print("    if nullptr: same as \"cdmox@B\"\n");
+  tty->print("    if null: same as \"cdmox@B\"\n");
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.cpp:2076]: null better here imho

File: src/hotspot/share/opto/node.hpp [hunk 5]
```diff
@@ -254,7 +254,7 @@ class Node {
 
   // Create a new Node with given input edges.
   // This version requires use of the "edge-count" new.
-  // E.g.  new (C,3) FooNode( C, NULL, left, right );
+  // E.g.  new (C,3) FooNode( C, null, left, right );
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.hpp:257]: nullptr

File: src/hotspot/share/opto/node.hpp [hunk 6]
```diff
@@ -841,7 +841,7 @@ class Node {
     return (type##Node*)this;                                \
   }                                                          \
   type##Node* isa_##type() const {                           \
-    return (is_##type()) ? as_##type() : NULL;               \
+    return (is_##type()) ? as_##type() : nullptr;               \
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.hpp:844]: align

File: src/hotspot/share/opto/node.hpp [hunk 7]
```diff
@@ -1198,7 +1198,7 @@ class Node {
   Node* find(int idx, bool only_ctrl = false); // Search the graph for the given idx.
   Node* find_ctrl(int idx); // Search control ancestors for the given idx.
   void dump_bfs(const int max_distance, Node* target, const char* options) const; // Print BFS traversal
-  void dump_bfs(const int max_distance) const; // dump_bfs(max_distance, nullptr, nullptr)
+  void dump_bfs(const int max_distance) const; // dump_bfs(max_distance, null, null)
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.hpp:1201]: nullptr

File: src/hotspot/share/opto/node.hpp [hunk 8]
```diff
@@ -1258,7 +1258,7 @@ class Node {
 };
 
 inline bool not_a_node(const Node* n) {
-  if (n == NULL)                   return true;
+  if (n == nullptr)                   return true;
```
Comment (jdksjolen) [inline src/hotspot/share/opto/node.hpp:1261]: align

File: src/hotspot/share/opto/parse1.cpp [hunk 9]
```diff
@@ -1690,8 +1690,8 @@ void Parse::merge_common(Parse::Block* target, int pnum) {
       RegionNode *r = new RegionNode(edges+1);
       gvn().set_type(r, Type::CONTROL);
       record_for_igvn(r);
-      // zap all inputs to NULL for debugging (done in Node(uint) constructor)
-      // for (int j = 1; j < edges+1; j++) { r->init_req(j, NULL); }
+      // zap all inputs to null for debugging (done in Node(uint) constructor)
+      // for (int j = 1; j < edges+1; j++) { r->init_req(j, null); }
```
Comment (jdksjolen) [inline src/hotspot/share/opto/parse1.cpp:1694]: nullptr

File: src/hotspot/share/opto/phaseX.cpp [hunk 10]
```diff
@@ -99,15 +99,15 @@ Node *NodeHash::hash_find( const Node *n ) {
...
+    return nullptr;
   }
   uint key = hash & (_max-1);
   uint stride = key | 0x01;
   NOT_PRODUCT( _look_probes++ );
   Node *k = _table[key];        // Get hashed value
   if( !k ) {                    // ?Miss?
     NOT_PRODUCT( _lookup_misses++ );
-    return NULL;                // Miss!
+    return nullptr;                // Miss!
```
Comment (jdksjolen) [inline src/hotspot/share/opto/phaseX.cpp:110]: align

File: src/hotspot/share/opto/phaseX.cpp [hunk 11]
```diff
@@ -129,11 +129,11 @@ Node *NodeHash::hash_find( const Node *n ) {
     k = _table[key];            // Get hashed value
     if( !k ) {                  // ?Miss?
       NOT_PRODUCT( _lookup_misses++ );
-      return NULL;              // Miss!
+      return nullptr;              // Miss!
```
Comment (jdksjolen) [inline src/hotspot/share/opto/phaseX.cpp:132]: align

File: src/hotspot/share/opto/phaseX.cpp [hunk 12]
```diff
@@ -156,7 +156,7 @@ Node *NodeHash::hash_find_insert( Node *n ) {
     _table[key] = n;            // Insert into table!
     debug_only(n->enter_hash_lock()); // Lock down the node while in the table.
     check_grow();               // Grow table if insert hit limit
-    return NULL;                // Miss!
+    return nullptr;                // Miss!
```
Comment (jdksjolen) [inline src/hotspot/share/opto/phaseX.cpp:159]: align

File: src/hotspot/share/opto/phaseX.cpp [hunk 13]
```diff
@@ -1514,7 +1514,7 @@ void PhaseIterGVN::remove_globally_dead_node( Node *dead ) {
                 }
               }
             }
-          } // if (in != NULL && in != C->top())
+          } // if (in != null && in != C->top())
```
Comment (jdksjolen) [inline src/hotspot/share/opto/phaseX.cpp:1517]: nullptr

File: src/hotspot/share/opto/subnode.cpp [hunk 14]
```diff
@@ -1774,18 +1774,18 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
...
   //    // re-trying it again and again.
-  //    if( !phase->allow_progress() ) return NULL;
+  //    if( !phase->allow_progress() ) return null;
   //  #endif
   //    // Not valid for unsigned compare because of corner cases in involving zero.
   //    // For example, replacing "X-1 <u Y" with "X <=u Y" fails to throw an
   //    // exception in case X is 0 (because 0-1 turns into 4billion unsigned but
   //    // "0 <=u Y" is always true).
-  //    if( cmp->Opcode() == Op_CmpU ) return NULL;
+  //    if( cmp->Opcode() == Op_CmpU ) return null;
```
Comment (jdksjolen) [inline src/hotspot/share/opto/subnode.cpp:1788]: nullptr

File: src/hotspot/share/opto/superword.cpp [hunk 15]
```diff
@@ -50,29 +50,29 @@ SuperWord::SuperWord(PhaseIdealLoop* phase) :
...
-  _iv(NULL),                              // induction var
+  _nlist(arena(), 8, 0, nullptr),            // scratch list of nodes
+  _stk(arena(), 8, 0, nullptr),              // scratch stack of nodes
+  _lpt(nullptr),                             // loop tree node
+  _lp(nullptr),                              // CountedLoopNode
+  _pre_loop_end(nullptr),                    // Pre loop CountedLoopEndNode
+  _bb(nullptr),                              // basic block
+  _iv(nullptr),                              // induction var
   _race_possible(false),                  // cases where SDMU is true
   _early_return(true),                    // analysis evaluations routine
```
Comment (jdksjolen) [inline src/hotspot/share/opto/superword.cpp:77]: align

File: src/hotspot/share/opto/superword.cpp [hunk 16]
```diff
@@ -737,9 +737,9 @@ void SuperWord::find_adjacent_refs() {
           memops.push(s);
         }
         best_align_to_mem_ref = find_align_to_ref(memops, max_idx);
-        if (best_align_to_mem_ref == NULL) {
+        if (best_align_to_mem_ref == nullptr) {
           if (TraceSuperWord) {
-            tty->print_cr("SuperWord::find_adjacent_refs(): best_align_to_mem_ref == NULL");
+            tty->print_cr("SuperWord::find_adjacent_refs(): best_align_to_mem_ref == null");
```
Comment (jdksjolen) [inline src/hotspot/share/opto/superword.cpp:742]: nullptr?

File: src/hotspot/share/opto/superword.cpp [hunk 17]
```diff
@@ -4649,7 +4649,7 @@ void SWPointer::Tracer::offset_plus_k_4(Node* n) {
 void SWPointer::Tracer::offset_plus_k_5(Node* n, Node* _invar) {
   if(_slp->is_trace_alignment()) {
     print_depth(); tty->print_cr(" %d SWPointer::offset_plus_k: FAILED since another invariant has been detected before", n->_idx);
-    print_depth(); tty->print("  \\ %d SWPointer::offset_plus_k: _invar != NULL: ", _invar->_idx); _invar->dump();
+    print_depth(); tty->print("  \\ %d SWPointer::offset_plus_k: _invar != null: ", _invar->_idx); _invar->dump();
```
Comment (jdksjolen) [inline src/hotspot/share/opto/superword.cpp:4652]: nullptr?

File: src/hotspot/share/opto/type.hpp [hunk 18]
```diff
@@ -1692,7 +1692,7 @@ class TypeInstKlassPtr : public TypeKlassPtr {
 
   // Convenience common pre-built types.
   static const TypeInstKlassPtr* OBJECT; // Not-null object klass or below
-  static const TypeInstKlassPtr* OBJECT_OR_NULL; // Maybe-null version of same
+  static const TypeInstKlassPtr* OBJECT_OR_nullptr; // Maybe-null version of same
```
Comment (jdksjolen) [inline src/hotspot/share/opto/type.hpp:1695]: Macro glitch

**Review by jdksjolen — COMMENTED** [review]
Okay, so most of this stuff looks good. At some points there are conventions that I'm unfamiliar with, I've tried my best to comment those.

File: src/hotspot/share/opto/escape.cpp [hunk 19]
```diff
@@ -2341,31 +2341,31 @@ bool ConnectionGraph::is_oop_field(Node* n, int offset, bool* unsafe) {
   return (is_reference_type(bt) || bt == T_NARROWOOP);
 }
 
-// Returns unique pointed java object or NULL.
+// Returns unique pointed java object or null.
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/escape.cpp:2344]: No, not necessary.

File: src/hotspot/share/opto/gcm.cpp [hunk 20]
```diff
@@ -1206,10 +1206,10 @@ Block* PhaseCFG::hoist_to_cheaper_block(Block* LCA, Block* early, Node* self) {
   while (LCA != early) {
     LCA = LCA->_idom;         // Follow up the dominator tree
 
-    if (LCA == NULL) {
+    if (LCA == nullptr) {
       // Bailout without retry
       assert(false, "graph should be schedulable");
-      C->record_method_not_compilable("late schedule failed: LCA == NULL");
+      C->record_method_not_compilable("late schedule failed: LCA == null");
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/gcm.cpp:1212]: is

File: src/hotspot/share/opto/runtime.cpp [hunk 21]
```diff
@@ -1547,7 +1547,7 @@ address OptoRuntime::rethrow_C(oopDesc* exception, JavaThread* thread, address r
 #ifndef PRODUCT
   SharedRuntime::_rethrow_ctr++;               // count rethrows
 #endif
-  assert (exception != NULL, "should have thrown a NULLPointerException");
+  assert (exception != nullptr, "should have thrown a nullPointerException");
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/runtime.cpp:1550]: I changed it to NullPointerException, avoiding resolving for now.

File: src/hotspot/share/opto/stringopts.cpp [hunk 22]
```diff
@@ -127,7 +127,7 @@ class StringConcat : public ResourceObj {
   static Node* skip_string_null_check(Node* value) {
     // Look for a diamond shaped Null check of toString() result
     // (could be code from String.valueOf()):
-    // (Proj == NULL) ? "null":"CastPP(Proj)#NotNULL
+    // (Proj == null) ? "null":"CastPP(Proj)#Notnull
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/stringopts.cpp:130]: Also has a missing ". Should it be `CastPP(proj)#NotNull`?

**Comment by jdksjolen** [comment]
I have now applied the manual review fixes which I found.

File: src/hotspot/share/opto/phaseX.cpp [hunk 23]
```diff
@@ -2220,9 +2220,9 @@ Node *PhaseCCP::transform( Node *n ) {
     uint cnt = clone->req();
     for( uint i = 0; i < cnt; i++ ) {          // For all inputs do
       Node *input = clone->in(i);
-      if( input != NULL ) {                    // Ignore NULLs
+      if( input != nullptr ) {                    // Ignore nulls
```
Comment (vnkozlov) [inline src/hotspot/share/opto/phaseX.cpp:2223]: comment's alignment

File: src/hotspot/share/opto/subnode.cpp [hunk 24]
```diff
@@ -1774,18 +1774,18 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   //    // counter.  If they use the PRE-incremented counter, then the counter has
   //    // to be incremented in a private block on a loop backedge.
   //    if( du && du->cnt(this) && du->out(this)[0]->Opcode() == Op_CountedLoopEnd )
-  //      return NULL;
+  //      return null;
```
Comment (vnkozlov) [inline src/hotspot/share/opto/subnode.cpp:1777]: nullptr

File: src/hotspot/share/opto/subnode.cpp [hunk 25]
```diff
@@ -1774,18 +1774,18 @@ Node *BoolNode::Ideal(PhaseGVN *phase, bool can_reshape) {
...
   //  #ifndef PRODUCT
   //    // Do not do this in a wash GVN pass during verification.
   //    // Gets triggered by too many simple optimizations to be bothered with
   //    // re-trying it again and again.
-  //    if( !phase->allow_progress() ) return NULL;
+  //    if( !phase->allow_progress() ) return null;
```
Comment (vnkozlov) [inline src/hotspot/share/opto/subnode.cpp:1782]: nullptr

File: src/hotspot/share/opto/superword.cpp [hunk 26]
```diff
@@ -4649,7 +4649,7 @@ void SWPointer::Tracer::offset_plus_k_4(Node* n) {
 void SWPointer::Tracer::offset_plus_k_5(Node* n, Node* _invar) {
   if(_slp->is_trace_alignment()) {
     print_depth(); tty->print_cr(" %d SWPointer::offset_plus_k: FAILED since another invariant has been detected before", n->_idx);
-    print_depth(); tty->print("  \\ %d SWPointer::offset_plus_k: _invar != NULL: ", _invar->_idx); _invar->dump();
+    print_depth(); tty->print("  \\ %d SWPointer::offset_plus_k: _invar != nullptr: ", _invar->_idx); _invar->dump();
```
Comment (vnkozlov) [inline src/hotspot/share/opto/superword.cpp:4652]: `is not null`

File: src/hotspot/share/opto/type.cpp [hunk 27]
```diff
@@ -2678,18 +2678,18 @@ intptr_t TypePtr::get_con() const {
...
-  if (res->isa_ptr() == NULL) {
+  if (res->isa_ptr() == nullptr) {
     return res;
   }
 
   const TypePtr* res_ptr = res->is_ptr();
-  if (res_ptr->speculative() != NULL) {
-    // type->speculative() == NULL means that speculation is no better
+  if (res_ptr->speculative() != nullptr) {
+    // type->speculative() == null means that speculation is no better
```
Comment (vnkozlov) [inline src/hotspot/share/opto/type.cpp:2687]: `is null`

**Review by vnkozlov — COMMENTED** [review]
In general this looks good. I found few cases need to change.

File: src/hotspot/share/opto/connode.hpp [hunk 28]
```diff
@@ -75,7 +75,7 @@ class ConPNode : public ConNode {
 
   // Factory methods:
   static ConPNode* make(address con) {
-    if (con == NULL)
+    if (con == nullptr)
       return new ConPNode( TypePtr::NULL_PTR ) ;
     else
       return new ConPNode( TypeRawPtr::make(con) );
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/connode.hpp:81]: Also removed the extra spaces in function calls.

File: src/hotspot/share/opto/graphKit.cpp [hunk 29]
```diff
@@ -519,7 +519,7 @@ void GraphKit::uncommon_trap_if_should_post_on_exceptions(Deoptimization::DeoptR
       // Do not try anything fancy if we're notifying the VM on every throw.
       // Cf. case Bytecodes::_athrow in parse2.cpp.
       uncommon_trap(reason, Deoptimization::Action_none,
-                    (ciKlass*)NULL, (char*)NULL, must_throw);
+                    (ciKlass*)nullptr, (char*)nullptr, must_throw);
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/graphKit.cpp:522]: If the function is overloaded `foo(void* x); foo(int* x);` or templated `template<typename T> foo(T* x);` then you have to cast `nullptr`. Perhaps there are more cases. In this case, `uncommon_trap` is overloaded.

File: src/hotspot/share/opto/graphKit.cpp [hunk 30]
```diff
@@ -650,7 +650,7 @@ void GraphKit::builtin_throw(Deoptimization::DeoptReason reason) {
   // allocation time and code size, by drastically reducing the number
   // of in-edges on the call to the uncommon trap.
 
-  uncommon_trap(reason, action, (ciKlass*)NULL, (char*)NULL, must_throw);
+  uncommon_trap(reason, action, (ciKlass*)nullptr, (char*)nullptr, must_throw);
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/graphKit.cpp:653]: `uncommon_trap` overloaded

**Comment by jdksjolen** [comment]
Thank you for the review @vnkozlov, I've added your fixes now.

File: src/hotspot/share/opto/graphKit.cpp (hunk 29 above)
[the remark this answers is not in the document]
Comment (vnkozlov) [reply src/hotspot/share/opto/graphKit.cpp:522]: Thank you for explanation.

**Review by vnkozlov — APPROVED** [review]
Looks good.

**Comment by jdksjolen** [comment]
Changed an assert after Clang complained (Mac builds failed).

**Comment by jdksjolen** [comment]
Passes tier1, tier2.

**Review by vnkozlov — APPROVED** [review]
Update is good.

**Comment by vnkozlov** [comment]
Cross compilation for ARM in GHA failed with:
```
/home/runner/work/jdk/jdk/src/hotspot/cpu/arm/interpreterRT_arm.cpp:309:56: error: operands to '?:' have different types 'std::nullptr_t' and 'intptr_t' {aka 'int'}
  309 |       _toGP[_last_gp++] = (*(intptr_t*)from_addr == 0) ? nullptr : from_addr;
      |                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/home/runner/work/jdk/jdk/src/hotspot/cpu/arm/interpreterRT_arm.cpp:311:45: error: operands to '?:' have different types 'std::nullptr_t' and 'intptr_t' {aka 'int'}
  311 |       *_to++ = (*(intptr_t*)from_addr == 0) ? nullptr : from_addr;
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

File: src/hotspot/share/opto/multnode.cpp [hunk 31]
```diff
@@ -155,7 +155,7 @@ void ProjNode::dump_compact_spec(outputStream *st) const {
 //----------------------------check_con----------------------------------------
 void ProjNode::check_con() const {
   Node* n = in(0);
-  if (n == NULL)       return;  // should be assert, but NodeHash makes bogons
+  if (n == nullptr)       return;  // should be assert, but NodeHash makes bogons
```
Comment (JesperIRL) [inline src/hotspot/share/opto/multnode.cpp:158]: Align return.

File: src/hotspot/share/opto/node.cpp [hunk 32]
```diff
@@ -1033,36 +1033,36 @@ Node* Node::uncast_helper(const Node* p, bool keep_deps) {
...
   // Find a precedence edge to move
   uint i = _cnt;
-  while( in(i) != NULL ) {
+  while( in(i) != nullptr ) {
     if (in(i) == n) return; // Avoid spec violation: duplicated prec edge.
     i++;
   }
-  _in[i] = n;                                // Stuff prec edge over NULL
-  if ( n != NULL) n->add_out((Node *)this);  // Add mirror edge
+  _in[i] = n;                                // Stuff prec edge over null
```
Comment (JesperIRL) [inline src/hotspot/share/opto/node.cpp:1050]: Align comment.

File: src/hotspot/share/opto/node.hpp [hunk 33]
```diff
@@ -519,14 +519,14 @@ class Node {
...
+    Node *last = nullptr;
     for (; i < _max-1; ++i) {
       Node *next = _in[i+1];
-      if (next == NULL) break;
+      if (next == nullptr) break;
       last = next;
     }
     _in[gap] = last; // Move last slot to empty one.
-    _in[i] = NULL;   // NULL out last slot.
+    _in[i] = nullptr;   // null out last slot.
```
Comment (JesperIRL) [inline src/hotspot/share/opto/node.hpp:529]: Align comment.

File: src/hotspot/share/opto/output.cpp [hunk 34]
```diff
@@ -2155,9 +2155,9 @@ Node * Scheduling::ChooseNodeToBundle() {
 
 #ifndef PRODUCT
     if (_cfg->C->trace_opto_output())
-      tty->print("#   ChooseNodeToBundle: NULL\n");
+      tty->print("#   ChooseNodeToBundle: null\n");
```
Comment (JesperIRL) [inline src/hotspot/share/opto/output.cpp:2158]: User observable change(?) Do we care?

File: src/hotspot/share/opto/phaseX.cpp [hunk 35]
```diff
@@ -2289,7 +2289,7 @@ Node *PhaseCCP::transform_once( Node *n ) {
         NOT_PRODUCT( inc_constants(); )
       } else if( n->is_Region() ) { // Unreachable region
         // Note: nn == C->top()
-        n->set_req(0, NULL);        // Cut selfreference
+        n->set_req(0, nullptr);        // Cut selfreference
```
Comment (JesperIRL) [inline src/hotspot/share/opto/phaseX.cpp:2292]: Align comment.

**Review by JesperIRL — APPROVED** [review]
Looks good in general. There are a bunch of places where the code previously was intentionally aligned (comments, braces, instructions etc) where it's now not aligned anymore since nullptr has more characters than NULL. I have put notes on the places I noticed.

There are several strings that are modified, some seems to be part of logging and other text that is observable by a user. I'm not sure what our policy is around keeping logging / hs_err.log etc stable so just want to bring it up so that it's not a surprise to anyone.

File: src/hotspot/share/opto/memnode.cpp [hunk 36]
```diff
@@ -267,8 +267,8 @@ static Node *step_through_mergemem(PhaseGVN *phase, MergeMemNode *mmem,  const T
     }
     if( !consistent ) {
       st->print("alias_idx==%d, adr_check==", alias_idx);
-      if( adr_check == NULL ) {
-        st->print("NULL");
+      if( adr_check == nullptr ) {
+        st->print("null");
```
[the remark this answers is not in the document]
Comment (jdksjolen) [reply src/hotspot/share/opto/memnode.cpp:271]: The tty, so yes. Previous changes to logging have been accepted

[the remark this answers is not in the document]
Comment (vnkozlov) [reply src/hotspot/share/opto/memnode.cpp:271]: This code print additional information on `tty` before we fail the assert at line 277. It helps us with debugging.
Note, the code is under (!consistent) check so it prints this only when something failed.

**Review by vnkozlov — APPROVED** [review]
Update is good.

**Comment by jdksjolen** [comment]
Passes tier1. Integrating, thank you for the reviews.

/integrate

**Pull request closed without merging**